Block/Adminhtml/Timeline/Index.php

```diff
@@ -55,36 +56,39 @@ class Index extends Template
      */
     private $config;
 
+    /**
+     * @var CronDataService
+     */
+    private $cronDataService;
+
     public function __construct(
         Context $context,
-        Collection $collection,
         DateTime $dateTime,
         DateTimeConverter $converter,
         CronConfiguration $config,
+        CronDataService $cronDataService,
         array $data = []
     ) {
         $this->dateTime = $dateTime;
         $this->cronData = [];
         $this->converter = $converter;
         $this->config = $config;
+        $this->cronDataService = $cronDataService;
 
-        $this->initializeStartAndEndTime($collection);
+        $this->initializeStartAndEndTime();
```

Comment:
Running additional actions in a constructor can be time consuming if the object is not being used.